File: src/gestorAplicacion/usuario/Cuenta.java
package gestorAplicacion.usuario;

import gestorAplicacion.economia.Contable;
import gestorAplicacion.economia.Divisa;

import java.io.Serializable;


public abstract class Cuenta implements Serializable, Contable {
    
    /**
	 * 
	 */
	private static final long serialVersionUID = -5757213716524384183L;
	
	
	private int numeroCuenta;
	private Usuario usuario;
    private double saldo;
    private Divisa divisa;
	private String nombre;
	
    protected Cuenta(Usuario usuario, Divisa divisa, String nombre)  {
        saldo = 0;
        setUsuario(usuario);
        setDivisa(divisa);
		setNombre(nombre);
    }
	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
    public int getNumeroCuenta() {
		return numeroCuenta;
	}

	public void setNumeroCuenta(int numeroCuenta) {
		this.numeroCuenta = numeroCuenta;
	}
	
    public Usuario getUsuario() {
		return usuario;
	}

	public void setUsuario(Usuario usuario) {
		this.usuario = usuario;
	}

	public void depositar(double cantidad) {
        saldo = saldo + cantidad;
    }

    public boolean retirar(double cantidad) {
    	if(cantidad<=saldo) {
    		saldo = saldo - cantidad;
    		return true;
    	}else {
    		return false;
    	}
    }

    public double getSaldo() {
        return saldo;
    }
    
    public void setSaldo(double saldo) {
		this.saldo = saldo;
	}

	public Divisa getDivisa() {
		return divisa;
	}

	public void setDivisa(Divisa divisa) {
		this.divisa = divisa;
	}
    
}

File: Python/gestorAplicacion/usuario/Cuenta.py
class Cuenta:

    numeroCuenta = 0

    def __init__(self, usuario, divisa, nombre):
        self._numeroCuenta = Cuenta.numeroCuenta
        self._usuario = usuario
        self._saldo = 0
        self._divisa = divisa
        self._nombre = nombre
        Cuenta.numeroCuenta += 1

    def getNombre(self):
        return self._nombre

    def setNombre(self, nombre):
        self._nombre = nombre
        
    def getNumeroCuenta(self):
        return self._numeroCuenta

    def setNumeroCuenta(self, numeroCuenta):
        self._numeroCuenta = numeroCuenta

    def getUsuario(self):
        return self._usuario

    def setUsuario(self, usuario):
        self._usuario = usuario

    def depositar(self, cantidad):
        self._saldo = self._saldo + cantidad

    def retirar(self, cantidad):
        if cantidad<=self._saldo:
            self._saldo = self._saldo - cantidad
            return True
        else:
            return False

    def getSaldo(self):
        return self._saldo

    def setSaldo(self, saldo):
        self._saldo = saldo

    def getDivisa(self):
        return self._divisa

    def setDivisa(self, divisa):
        self._divisa = divisa

